android_lib/src/main/java/io/sentry/godotplugin/SentryAndroidGodotPlugin.kt

```diff
@@ -0,0 +1,376 @@
+import org.godotengine.godot.plugin.GodotPlugin
+import org.godotengine.godot.plugin.UsedByGodot
+import org.godotengine.godot.variant.Callable
+import org.threeten.bp.format.DateTimeParseException
+import kotlin.random.Random
+
+
+@Suppress("unused")
+class SentryAndroidGodotPlugin(godot: Godot) : GodotPlugin(godot) {
+    companion object {
+        private const val TAG = "sentry-godot"
+    }
+
+    private val eventsByHandle = ThreadLocal.withInitial { mutableMapOf<Int, SentryEvent>() }
+    private val exceptionsByHandle = ThreadLocal.withInitial { mutableMapOf<Int, SentryException>() }
+
+    private fun getEvent(eventHandle: Int): SentryEvent? {
+        val event: SentryEvent? = eventsByHandle.get()!![eventHandle]
+        if (event == null) {
+            Log.e(TAG, "Internal Error -- SentryEvent not found: $eventHandle")
+        }
+        return event
+    }
+
+    private fun getException(exceptionHandle: Int): SentryException? {
+        val exception: SentryException? = exceptionsByHandle.get()!![exceptionHandle]
+        if (exception == null) {
+            Log.e(TAG, "Internal Error -- SentryException not found: $exceptionHandle")
+        }
+        return exception
+    }
+
+    private fun registerEvent(event: SentryEvent): Int {
+        val eventsMap = eventsByHandle.get()!!
+        var handle = Random.nextInt()
+        while (eventsMap.containsKey(handle)) {
+            handle = Random.nextInt()
+        }
+        eventsMap[handle] = event
+        return handle
+    }
+
+    private fun registerException(exception: SentryException): Int {
+        val exceptionsMap = exceptionsByHandle.get()!!
+        var handle = Random.nextInt()
+        while (exceptionsMap.containsKey(handle)) {
+            handle = Random.nextInt()
+        }
+        exceptionsMap[handle] = exception
+        return handle
+    }
+
+    override fun getPluginName(): String {
+        return "SentryAndroidGodotPlugin"
+    }
+
+    override fun onGodotSetupCompleted() {
+        super.onGodotSetupCompleted()
+        AndroidThreeTen.init(this.activity!!.application) // needed for date time zones
+    }
+
+    @UsedByGodot
+    fun initialize(
+        beforeSendHandlerId: Long,
+        dsn: String,
+        debug: Boolean,
+        release: String,
+        dist: String,
+        environment: String,
+        sampleRate: Float,
+        maxBreadcrumbs: Int,
+    ) {
+        Log.v(TAG, "Initializing Sentry Android")
+        SentryAndroid.init(godot.getActivity()!!.applicationContext) { options ->
+            options.dsn = dsn.ifEmpty { null }
+            options.isDebug = debug
+            options.release = release.ifEmpty { null }
+            options.dist = dist.ifEmpty { null }
+            options.environment = environment.ifEmpty { null }
+            options.sampleRate = sampleRate.toDouble()
+            options.maxBreadcrumbs = maxBreadcrumbs
+            options.sdkVersion!!.name = "sentry.java.android.godot"
```

Comment:
Note we also need to add `sentry.native.android.godot` at the NDK layer
e.g:

https://github.com/getsentry/sentry-unreal/blob/e2a7b64f31ab0a3d8b43c7a36b686d9351a819a9/scripts/build-android.sh#L10

https://github.com/getsentry/sentry-unity/blob/c97e63f04fcb559f3740556041f8a3b497bbec3c/Directory.Build.targets#L239